### PR TITLE
Updated for SDL3 changes to gamepad face buttons

### PR DESCRIPTION
**Pull request by slouken**
Updated for SDL3 changes to gamepad face buttons

### CODE REVIEW

File: src/sdl2_compat.c [hunk 1]
```diff
@@ -1350,6 +1352,79 @@ SDL_LOG_IMPL(Error, ERROR)
 SDL_LOG_IMPL(Critical, CRITICAL)
 #undef SDL_LOG_IMPL
 
+/* !!! FIXME: when we override SDL_Quit(), we need to free/reset gamepad_button_swap_list */
```
Comment (sezero) [inline src/sdl2_compat.c:1355]: We override SDL_Quit already (for gesture stuff)

Comment (slouken) [reply src/sdl2_compat.c]: Great, I added cleanup for that.

Comment (sezero) [reply src/sdl2_compat.c]: Should we do that cleanup after SDL3_Quit like you do, or before?

File: src/sdl2_compat.c [hunk 2]
```diff
@@ -3544,6 +3623,24 @@ SDL_Quit(void)
...
     SDL3_Quit();
+
+    if (joystick_list) {
+        SDL_free(joystick_list);
+        joystick_list = NULL;
+    }
+    num_joysticks = 0;
+
+    if (sensor_list) {
+        SDL_free(sensor_list);
```
Comment (sezero) [inline src/sdl2_compat.c:3634]: `SDL3_free` instead of `SDL_free` (same for the ones below)

Comment (slouken) [reply src/sdl2_compat.c]: Fixed

File: src/sdl2_compat.c (hunk 1 above)
Comment (slouken) [reply src/sdl2_compat.c]: My intuition is that there may be events in SDL_Quit() and we might need the data there, but I'll move it beforehand for consistency with everything else for now.

Comment (sezero) [reply src/sdl2_compat.c]: OK then

**Review by sezero — APPROVED** [review]

**Pull request merged**